syntax/nix.vim

```diff
@@ -50,6 +50,7 @@ syn region nixString matchgroup=nixStringDelimiter start=+''+ skip=+''['$\\]+ en
 syn match nixFunctionCall "[a-zA-Z_][a-zA-Z0-9_'-]*"
 
 syn match nixPath "[a-zA-Z0-9._+-]*\%(/[a-zA-Z0-9._+-]\+\)\+"
+syn region nixInterpolatedPath start="\%(\~\|[a-zA-Z0-9._+-]*\)\%(\/[a-zA-Z0-9._+/-]\+\|\/\)\ze\%(${\)" skip="[a-zA-Z0-9._+/-]*\ze${" end="\%(}\)\@<=[a-zA-Z0-9._+/-]*\%([^a-zA-Z0-9._+/-]\|$\)\@=" contains=nixInterpolation oneline
```

Comment:
`oneline` is not enough, because you can have something like this:

```nix
foo/${
  "bar"
}
```

Test case:

```vader
Given nix (path-with-multiline-subexpr):
  foo/${
    "bar"
  }

Execute (syntax):
  AssertEqual SyntaxOf('foo/'), 'nixInterpolatedPath'
  AssertEqual SyntaxOf('${'), 'nixInterpolationDelimiter'
  AssertEqual SyntaxOf('"'), 'nixStringDelimiter'
  AssertEqual SyntaxOf('bar'), 'nixSimpleString'
  AssertEqual SyntaxOf('}'), 'nixInterpolationDelimiter'
```

Reply:
Very good catch, thanks! Pushed a fix.